src/SheetsRegistry.js

```diff
@@ -15,7 +15,21 @@ export default class SheetsRegistry {
    * @api public
    */
   add(sheet) {
-    this.registry.push(sheet)
+    const {registry} = this
+
+    if (registry.length && sheet.options.index) {
+      for (let i = 0; i < registry.length; i++) {
+        const registration = registry[i]
+        if (
+          !registration.options.index ||
+          registration.options.index < sheet.options.index
+        ) continue
+        registry.splice(i, 0, sheet)
+        return
+      }
+    }
+
+    registry.push(sheet)
   }
```

Comment:
Here is an untested sample of the same code, how I would write it, I think it is a bit easier to read because of early returns, less indentation, less conditions:

```js

const {registry} = this

if (!registry.length || sheet.options.index != null) {
  registry.push(sheet)
  return
}

for (let i = 0; i < registry.length; i++) {
  const {options} = registry[i]
  if (options.index >= sheet.options.index) {
    registry.splice(i, 0, sheet)
    return
  } 
}

```